sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -616,18 +616,22 @@ booleanValue
     ;
 
 interval
-    : INTERVAL intervalField*
+    : INTERVAL? intervalField+
     ;
 
 intervalField
-    : value=intervalValue unit=identifier (TO to=identifier)?
+    : value=intervalValue unit=intervalUnit (TO to=identifier)?
```

Comment:
shouldn't the `to` also a intervalUnit?

Reply:
right, I checked the hive parser and it seems to handle `TO` as units. But, this is an refactoring issue, so is it okay to include this fix in this pr?